src/Bicep.Core/CoreResources.resx

```diff
@@ -169,11 +172,54 @@
   </data>
   <data name="NoUnnecessaryDependsOnRuleDescription" xml:space="preserve">
     <value>No unnecessary dependsOn.</value>
+    <comment />
   </data>
   <data name="NoUnnecessaryDependsOnRuleMessage" xml:space="preserve">
     <value>Remove unnecessary dependsOn entry '{0}'.</value>
     <comment>Text for linter rule error. {0} is a programmatic expression</comment>
   </data>
+  <data name="NoHardcodedLocationRuleDescription" xml:space="preserve">
+    <value>A resource's location should not use a hard-coded string or variable value. It should use a parameter, an expression, or the string 'global'.</value>
+    <comment />
+  </data>
+  <data name="NoLocExprOutsideParamsRuleError" xml:space="preserve">
+    <value>Use a parameter here instead of '{0}'. 'resourceGroup().location' and 'deployment().location' should only be used as a default value for parameters.</value>
+    <comment>{0} = programmatic expression, e.g. 'resourceGroup().location'</comment>
+  </data>
+  <data name="NoLocExprOutsideParamsRuleDescription" xml:space="preserve">
+    <value>Functions resourceGroup().location and deployment().location should only be used as the default value of a parameter.</value>
+    <comment>{0} = programmatic expression, e.g. 'resourceGroup().location'</comment>
+  </data>
+  <data name="NoHardcodedLocation_ErrorChangeVarToParam" xml:space="preserve">
+    <value>A resource location should not use a hard-coded string or variable value. Change variable '{0}' into a parameter.</value>
+    <comment>{0} is a variable name</comment>
+  </data>
+  <data name="NoHardcodedLocation_FixChangeVarToParam" xml:space="preserve">
+    <value>Change variable '{0}' into a parameter</value>
+    <comment>{0} is a variable name</comment>
+  </data>
+  <data name="NoHardcodedLocation_FixNewParam" xml:space="preserve">

```

Comment:
Not a revision, just wondering what is "fix new param" versus "fix change var to new param"

Reply:
Two fixes available:
1) change existing variable to a param (if applicable)
2) add a new param

Reply:
I don't currently have implemented a way to change the expression to an existing parameter.

Reply:
Okay, that is fine, that is an auto-fix to change an expression to an existing parameter